### DISCUSSION TITLE
Key encryption lost after downgrading 1.2.10 to 1.2.9

### DISCUSSION THREAD

**Discussion by FFurno** [Chia-Network/chia-blockchain#8998] [Support]
Hi all, I had initially installed 1.20.9 on a new windows machine. I then upgraded to 1.2.10 and encrypted the keys with the new functionality. However, since there is a bug with the pool address field (already pointed out), I downgraded to 1.2.9 to join a pool and get a pool contract for plotting. 

However, when after installing 1.2.9, the encryption was gone. I would like to understand if this is unintended behavior, or if I am supposed to manually delete keys from the previous location after upgrading to 1.2.10. Thank you

**Comment by shaneo257** [comment]
1.2.9 doesn't have the key encryption that 1.2.10 has

**Reply by FFurno** [reply]
Of course, but then I should not be able to access my encrypted private keys when I downgrade to 1.2.9. So either there is an extra set of private keys that needs to be removed manually, or the encryption is not working as expected

**Comment by alexharrington** [comment]
See the Additional Notes section of https://github.com/Chia-Network/chia-blockchain/wiki/Passphrase-Protected-Chia-Keys-and-Key-Storage-Migration

It says that the old keys are deliberately left in place to facilitate a downgrade. I'm not an expert so I don't know where that would be on a Windows PC I'm afraid, but yes you would need to securely delete those files once you upgrade again I think.

**Reply by FFurno** [reply]
I missed it. It makes sense, they could make it a bit more explicit during the migration phase. Thank you!

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
